### ISSUE TITLE
Students unable to proceed if resuming an attempt for a module that was updated

### ISSUE THREAD

**Issue by zachberry** [ucfopen/Obojobo#1587]
I noticed this on my numeric branch, which may be the only branch experiencing the issue, but I believe it should be happening to master now. I think there's an issue with the following:

1. Student: Start an attempt on Draft A, version 1. Don't complete the attempt
2. Instructor: Update Draft A, which is now Draft A, version 2.
3. Student: Return to Draft A to resume your attempt. When submitting you get the following error:

![image](https://user-images.githubusercontent.com/73479/95473837-7df35780-0952-11eb-93d2-d286af155053.png)

Due to the following code in attempt-end.js:

```javascript
if (
	req.currentDocument.draftId !== attempt.draftId ||
	req.currentDocument.contentId !== attempt.draftContentId
) {
	throw Error('Cannot end an attempt for a different module')
}
```

The problem is that once you get this message you can never proceed, you'll always get this message.

My first thought on how to solve this - When saving a module we do a check for any in-progress attempts. If any are found we warn the author that by changing this module they'll invalidate X attempts. If they agree, then any in-progress attempts are deleted. Should a student return to the module they won't be forced to resume the attempt and will have to start the attempt over. This would prevent them from getting stuck.

**Comment by vutoan1245** [comment]
My solution:

Make 2 new API endpoints
1. Check for any in-progress attempt
2. Delete all in-progress attempts

When saving, call the API 1
- If there is none, continue saving
- Otherwise, display a modal stating if the user wants to delete it 

What do you think? @iturgeon @zachberry

**Comment by iturgeon** [comment]
The biggest problem I see with this solution is that we'll be deleting attempts that are sitting at rest while a student is away AND attempts that are actively being taken. 

The active ones' experience is going to be pretty bad.

I thought it was already implemented like this, if not... Thinking it should be.

Visit attaches user, attempts, and doc version together. An open attempt in a visit stays attached to the specific version of the document no matter if there's a newly saved doc. Care needs to be taken to use the potentially older version by using the linked content I'd from the visit.

If the student starts an attempt, closes, then returns after a new version is saved. The user's open attempt will be for an old version and is not resumable - must be ignored/reset/maybe deleted.

**Comment by zachberry** [comment]
I found this on production with two of Dr. Chen's students

**Comment by zachberry** [comment]
@iturgeon and I discussed how we want to proceed. Our conclusion is we'll throw the now out-of-date attempt out and the student will need to start over. 

* When a student returns to a module with an unfinished attempt, but that unfinished attempt was for a different draft_content_id (aka for an older version of the module), they'll be presented with the same UI (You were in the middle of an attempt, let's continue...) but continuing will actually start a whole new attempt.
* If a student happens to be in the middle of an attempt while the module is updated, attempting to submit the assessment currently errors (and for now, whitescreens the viewer). That error should be caught instead, with a dialog that says that the module has been updated and that the student needs to refresh the page. We'll let them close the dialog since it's possible that the new module has some or all of the same questions, and they may want to write down their answers. A first draft of the dialog text:

```
This module was edited since you started your attempt. Refresh the page to start a new attempt with the updated module, or close this dialog to review your answers before you refresh.

[ Close ] [ Refresh Page ]
```
